catalogue/webapp/pages/images.tsx

```diff
@@ -0,0 +1,283 @@
+  const { searchPrototype } = useContext(TogglesContext);
+
+  const { query, page } = imagesRouteProps;
+
+  useEffect(() => {
+    trackSearch(apiProps, {
+      totalResults: results?.type === 'ResultList' ? results?.totalResults : 0,
+      source: Router.query.source || 'unspecified',
+    });
+  }, [imagesRouteProps]);
+
+  useEffect(() => {
+    function routeChangeStart(url: string) {
+      setLoading(true);
+    }
+    function routeChangeComplete(url: string) {
+      setLoading(false);
+    }
+    Router.events.on('routeChangeStart', routeChangeStart);
+    Router.events.on('routeChangeComplete', routeChangeComplete);
+
+    return () => {
+      Router.events.off('routeChangeStart', routeChangeStart);
+      Router.events.off('routeChangeComplete', routeChangeComplete);
+    };
+  }, []);
+
+  useHotjar();
+
+  if (results && results.type === 'Error') {
+    return (
+      <ErrorPage
+        title={
+          results.httpStatus === 500
+            ? `We're experiencing technical difficulties at the moment. We're working to get this fixed.`
+            : undefined
+        }
+        statusCode={results.httpStatus}
+      />
+    );
+  }
+
+  return (
+    <>
+      <Head>
+        {results?.type === 'ResultList' && results.prevPage && (
+          <link
+            rel="prev"
+            href={convertUrlToString(
+              imagesLink(
+                { ...imagesRouteProps, page: (page || 1) - 1 },
+                'meta_link'
+              ).as
+            )}
+          />
+        )}
+        {results?.type === 'ResultList' && results.nextPage && (
+          <link
+            rel="next"
+            href={convertUrlToString(
+              imagesLink({ ...imagesRouteProps, page: page + 1 }, 'meta_link')
+                .as
+            )}
+          />
+        )}
+      </Head>
+
+      <CataloguePageLayout
+        title={`${query ? `${query} | ` : ''}image search`}
+        description="Search Wellcome Collection images"
+        url={imagesLink({ ...imagesRouteProps }, 'canonical_link').as}
+        openGraphType={'website'}
+        jsonLd={{ '@type': 'WebPage' }}
+        siteSection={'collections'}
+        imageUrl={null}
+        imageAltText={null}
+      >
+        <Space
+          v={{
+            size: 'l',
+            properties: ['padding-bottom'],
+          }}
+          className={classNames(['row'])}
+        >
+          <div className="container">
+            <div className="grid">
+              <div className={grid({ s: 12, m: 12, l: 12, xl: 12 })}>
+                {searchPrototype ? (
+                  <>
+                    <SearchTabs
+                      worksRouteProps={imagesRoutePropsToWorksRouteProps(
+                        imagesRouteProps
+                      )}
+                      imagesRouteProps={imagesRouteProps}
+                      workTypeAggregations={[]}
+                      shouldShowImagesFilters={query !== ''}
```

Comment:
Could this logic not exist in the component? Or are passing these props from somewhere else too?

Reply:
The need for it has gone away entirely now we're changing the search context immediately when you switch tabs, so we can get rid of it.